collectors/ebpf.plugin/ebpf_disk.c

```diff
@@ -0,0 +1,819 @@
+}
+
+/**
+ * Compare disks
+ *
+ * Compare major and minor values to add disks to tree.
+ *
+ * @param a pointer to netdata_ebpf_disks
+ * @param b pointer to netdata_ebpf_disks
+ *
+ * @return It returns 0 case the values are equal, 1 case a is bigger than b and -1 case a is smaller than b.
+*/
+static int ebpf_compare_disks(void *a, void *b)
+{
+    netdata_ebpf_disks_t *ptr1 = a;
+    netdata_ebpf_disks_t *ptr2 = b;
+
+    if (ptr1->dev > ptr2->dev)
+        return 1;
+    if (ptr1->dev < ptr2->dev)
+        return -1;
+
+    return 0;
+}
+
+/**
+ * Update listen table
+ *
+ * Update link list when it is necessary.
+ *
+ * @param name  the    disk name
+ * @param major major  disk identifier
+ * @param minor minor  disk identifier
+ * @param current_time current timestamp
+ */
+static void update_disk_table(char *name, int major, int minor, time_t current_time)
+{
+    netdata_ebpf_disks_t find;
+    netdata_ebpf_disks_t *w;
+    size_t length;
+
+    uint32_t dev = netdata_new_encode_dev(major, minor);
+    find.dev = dev;
+    netdata_ebpf_disks_t *ret = (netdata_ebpf_disks_t *) avl_search_lock(&disk_tree, (avl_t *)&find);
+    if (ret) { // Disk is already present
+        ret->flags |= NETDATA_DISK_IS_HERE;
+        ret->last_update = current_time;
+        return;
+    }
+
+    netdata_ebpf_disks_t *update_next = disk_list;
+    if (likely(disk_list)) {
+        netdata_ebpf_disks_t *move = disk_list;
+        while (move) {
+            if (dev == move->dev)
+                return;
+
+            update_next = move;
+            move = move->next;
+        }
+
+        w = callocz(1, sizeof(netdata_ebpf_disks_t));
+        length = strlen(name);
+        if (length >= NETDATA_DISK_NAME_LEN)
+            length = NETDATA_DISK_NAME_LEN;
+
+        strncpy(w->family, name, length);
+        w->family[length] = '\0';
+        w->major = major;
+        w->minor = minor;
+        w->dev = netdata_new_encode_dev(major, minor);
+        update_next->next = w;
+    } else {
+        disk_list = callocz(1, sizeof(netdata_ebpf_disks_t));
+        length = strlen(name);
+        if (length >= NETDATA_DISK_NAME_LEN)
+            length = NETDATA_DISK_NAME_LEN;
+
+        strncpy(disk_list->family, name, length);
+        disk_list->family[length] = '\0';
+        disk_list->major = major;
+        disk_list->minor = minor;
+        disk_list->dev = netdata_new_encode_dev(major, minor);
+
+        w = disk_list;
+    }
+
+    ebpf_read_disk_info(w, name);
+
+    netdata_ebpf_disks_t *check;
+    check = (netdata_ebpf_disks_t *) avl_insert_lock(&disk_tree, (avl_t *)w);
+    if (check != w)
+        error("Internal error, cannot insert the AVL tree.");
+
+#ifdef NETDATA_INTERNAL_CHECKS
+    info("The Latency is monitoring the hard disk %s (Major = %d, Minor = %d, Device = %u)", name, major, minor,w->dev);
+#endif
+
+    w->flags |= NETDATA_DISK_IS_HERE;
+}
+
+/**
+ *  Read Local Ports
```

Comment:
Ports?

Reply:
I fixed the minor problems with the last commits now I will work with trace reported.